### PR TITLE
String manipulation 2

### PR DESCRIPTION
**Pull request by DreSean-Sutton**
image

### CODE REVIEW

File: string-manipulation-2/src/swap-chars.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+...as it's argument of the slice method of str with 0 as it's argument
+-create a variable named firstSwap and assign it the value of newStr[firstIndex]
+-create a variable named secondSwap and assign it the value of newStr[secondIndex]
+-call the splice method of newStr with firstIndex, 1, secondSwap as it's arguments
+-call the splice method of newStr with secondIndex, 1, firstSwap as it's arguments
+-return the join method of newStr with '' as it's arguments
+*/
+
+function swapChars(firstIndex, secondIndex, str) {
+  var newStr = str.slice(0).split('');
```
Comment (codingcodymiller) [inline string-manipulation-2/src/swap-chars.js]: Try doing this problem without converting the string to an array 🤓

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by DreSean-Sutton** [comment]
image

File: string-manipulation-2/src/num-vowels.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    if ((str[i] === 'a') ||
+        (str[i] === 'A') ||
+        (str[i] === 'e') ||
+        (str[i] === 'E') ||
+        (str[i] === 'i') ||
+        (str[i] === 'I') ||
+        (str[i] === 'o') ||
+        (str[i] === 'O') ||
+        (str[i] === 'u') ||
+        (str[i] === 'U')) {
```
Comment (thebearingedge) [inline string-manipulation-2/src/num-vowels.js]: If you lowercase your string you can check half as many comparisons.

```js
var lowerStr = str.toLowerCase();
```

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Comment by DreSean-Sutton** [comment]
image

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**